manticore/core/state.py

```diff
@@ -461,20 +461,31 @@ def solve_one_n(self, *exprs, constrain=False):
         :return: Concrete value or a tuple of concrete values
         :rtype: int
         """
-        values = []
-        for expr in exprs:
-            if not issymbolic(expr):
-                values.append(expr)
-            else:
-                expr = self.migrate_expression(expr)
-                value = self._solver.get_value(self._constraints, expr)
-                if constrain:
-                    self.constrain(expr == value)
-                # Include forgiveness here
-                if isinstance(value, bytearray):
-                    value = bytes(value)
-                values.append(value)
-        return values
+        return self.solve_one_n_batched(exprs, constrain)
+
+    def solve_one_n_batched(self, exprs, constrain=False):
+        """
+        Concretize a symbolic :class:`~manticore.core.smtlib.expression.Expression` into
+        one solution.
+        :param exprs: An iterable of manticore.core.smtlib.Expression
+        :param bool constrain: If True, constrain expr to solved solution value
```

Comment:
To be honest, I haven't look on that parameter and I haven't change how other code call this. If you say this is expected behavior, I will modify the comment.

Reply:
Right, this is the expected behavior, but the comment makes no mention of it. This could be confusing to users because if you call `solve_one_n_batched` with contradictory constraints in `exprs`, which expressions are `unsat` will depend on which ones are solved first. It's been that way for as long as I can remember, so this is just a "as long as you're editing this function anyway..." suggestion